### PR TITLE
Documentation updates

### PR DESCRIPTION
**Pull request by zacsimile**
Addresses issue #persistent need for better documentation.

**Is this a bugfix or an enhancement?**
Enhancement

**Proposed changes:**
- Add documentation guidelines to contributing
- Update installation instructions for clarity
- Update doc copyright

**Checklist:**

- [ ] Tested with numpy=1.14
- [ ] Tested on python 2.7 and 3.6
- [ ] Tested with wx=3.x and wx=4.x [if UI code]
- [ ] Does the PR avoid variable renaming in existing code, whitespace changes, and other forms of tidying? [There is a place for code tidying, but it makes reviewing 
much simpler if this is kept separate from functional changes]

If an enhancement (or non-trivial bugfix):

- [ ] Has this been discussed in advance (feature request, PR proposal, email, or direct conversation)?
- [ ] Does this change how users interact with the software? How will these changes be communicated?
- [ ] Does this maintain backwards compatibility with old data?
- [ ] Does this change the required dependencies?
- [ ] Are there any other side effects of the change?

### CODE REVIEW

File: docs/Installation/Installation.rst [hunk 1]
```diff
@@ -1,10 +1,129 @@
...
+
+conda installation for users
+============================
+
+Download and install `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_.
+Then, open a command prompt and enter
+
+.. code-block:: bash
+	
+    conda create -n pyme python=2.7
```
Comment (barentine) [inline docs/Installation/Installation.rst]: I don't think we should old up Zach's doc-update for this, but maybe we should revisit this week whether there's anything left to do before we make py3 the default installation. If the answer is just packaging and docs (notably launchWorkers -> PYMEClusterOfOne) I think it'd be great if we push through on that this week - I'll have a go at the launchWorkers -> PYMEClusterOfOne.

**Comment by zacsimile** [comment]
Worth noting that the `pip` install for users doesn't work.

```
(base) Mor:python-microscopy zachcm$ conda create -n pyme_pip python=2.7
(base) Mor:python-microscopy zachcm$ conda activate pyme_pip
(pyme_pip) Mor:python-microscopy zachcm$ pip install python-microscopy
DEPRECATION: Python 2.7 will reach the end of its life on January 1st, 2020. Please upgrade your Python as Python 2.7 won't be maintained after that date. A future version of pip will drop support for Python 2.7. More details about Python 2 support in pip, can be found at https://pip.pypa.io/en/latest/development/release-process/#python-2-support
ERROR: Could not find a version that satisfies the requirement python-microscopy (from versions: none)
ERROR: No matching distribution found for python-microscopy
```

I followed the link in the deprecation warning and it looks like `pip` will drop Python 2.7 support in January 2021. This prompted me to check `conda`, which has apparently already dropped Py2 support (https://docs.anaconda.com/anaconda/reference/release-notes/). Users will need to install an old version of Anaconda going forward to ensure they can use our Py2 branch. Right now, Miniconda install page still links to this old version, but it's something we'll need to be aware of.

The `conda` install for users also doesn't work on Mac because of a framework build issue. If I `conda-build python-microscopy` locally and then install it, the framework issue disappears. I'm not sure if this is because I'm compiling on my machine, or because the uploaded package is compiled incorrectly. Worth trying a rebuild using the latest version of `conda-build` and the latest packaging scripts and re-uploading to the `david_baddeley` channel.

File: docs/Installation/Installation.rst (hunk 1 above)
Comment (David-Baddeley) [reply docs/Installation/Installation.rst]: Our target for making Py3 the default is September. Before then I'd like to a) have @csoeller try it out and b) update the analysis docs etc for `PYMEClusterOfOne`. I'd be open, however, to making sure that packages for py3 are built and adding an installing on py3 section / note to the installation docs.

Comment (zacsimile) [reply docs/Installation/Installation.rst]: These new docs tell users to install Py2 and developers to install Py3. I think this, coupled with joining image.sc, should help with beta testing Py3. I think we're far along enough to expect developers to make it through any Py3 problems. There should be none on the initial install or for general usage. Users in our lab are on Py3 with no issues. I think it's worth updating the analysis docs for the developers and see how they get on.

File: docs/Installation/InstallationFromSource.rst [hunk 2]
```diff
@@ -7,7 +7,7 @@ If you want to contribute to PYMEs development, or you want to use PYME in a con
 requires modification to some of the scripts you will want to have the PYME sources somewhere you can easily find, modify
 and update them.
 
-If you just want to use the data analysis features then you are strongly advised to follow the instructions in :ref:`installationanaconda`_
```
Comment (David-Baddeley) [inline docs/Installation/InstallationFromSource.rst:10]: Why delete the underscore? I'm pretty sure it's need to turn this into a link.

File: docs/Installation/Installation.rst [hunk 3]
```diff
@@ -1,10 +1,129 @@
...
+.. code-block:: bash
+
+    cd /path/to/python-microscopy/
+    /path/to/mininconda/install/python.app/Contents/MacOS/python setup.py develop
+
+Additional resources
+********************
+
+- Detailed developer installation docs are located at :ref:`installationFromSource`
+- Additional PYME installation docs can be found at https://github.com/csoeller/pyme-install-docs
```
Comment (David-Baddeley) [inline docs/Installation/Installation.rst]: If there is something missing from our install docs, we should fix them - outsourcing our install docs is just plain weird.

**Comment by David-Baddeley** [comment]
I'm not happy with the proposed changes to the install docs - for an end user it's not OK to present 6 different options. We need to present the one option which we want them to use. Ideally it would be the executable installer, but as a) that currently doesn't get built regularly b) doesn't work on OSX it's miniconda etc ....

File: docs/Installation/InstallationFromSource.rst (hunk 2 above)
Comment (zacsimile) [reply docs/Installation/InstallationFromSource.rst:10]: For external links you need the underscore, for internal ones you need to remove it. Verified by compiling the docs.

**Comment by zacsimile** [comment]
Right now, if you go to our install docs you have to read several paragraphs before you can do anything. Every step is interspersed with multiple notes and warnings. This is more than a user wants to consider when first installing a program. By moving all caveats below the install instructions, you can quickly install PYME (assuming the packages work, which is addressed in #321, python-microscopy/pyme-conda-recipes#7 and python-microscopy/pyme-conda-recipes#8) without having to think about any of the steps. I’ve listed our options in a way I think makes it easy to identify relevance to the reader. Two options are for users, so users will follow one of those. Users who don’t know what to do will follow the first option, as they should. Developers should be fine following any of these instructions.

I think multiple installation options is OK/common. Numpy presents two options: https://numpy.org/install/. scipy, Cell Profiler and napari present multiple options: https://scikit-image.org/docs/stable/install.html, https://github.com/CellProfiler/CellProfiler/blob/master/README.md, https://github.com/napari/napari/blob/master/README.md. I tried to make the new install instructions I propose here similar to the install instructions from these packages since they all have a lot of users.

How do you propose presenting our multiple installation routes in a user-friendly manner?

File: docs/Installation/Installation.rst (hunk 3 above)
Comment (zacsimile) [reply docs/Installation/Installation.rst]: It was a nod to Christian and his work. We can remove it.

**Comment by barentine** [comment]
@zacsimile with the merge of #320 you should no longer need to specify docutils for pip install. That said, until #352 is merged sphinx needs to be specified (for bakeshop to work). 

If mpld3 is required to run PYMEVisualize I think you should submit a PR to add it into requirements.txt

File: docs/Installation/Installation.rst (hunk 1 above)
Comment (David-Baddeley) [reply docs/Installation/Installation.rst]: I'm a little reluctant to tell *users* to install in an environment as (at least on windows) I've run into issues running programs from shortcuts and/or associating files with programs this way. The environment way pretty much forces people to drop into an anaconda prompt and run `conda activate pyme` before running stuff on the command line. 

Installing in the base environment avoids these issues.

File: docs/Installation/Installation.rst [hunk 4]
```diff
@@ -3,8 +3,141 @@
...
+
+Installing using conda
+======================
+
+Download and install the python **2.7** version of `Miniconda <https://docs.conda.io/en/latest/miniconda.html>`_.
+Then, open the *Anaconda prompt* [#anacondaprompt]_ and enter
+
+.. code-block:: bash
+	
+    conda config --add channels david_baddeley
```
Comment (David-Baddeley) [inline docs/Installation/Installation.rst:25]: Do we not need the `anaconda` channel?

Comment (zacsimile) [reply docs/Installation/Installation.rst:25]: The last time I nuked and paved the `anaconda` channel was there by default. Probably safer to add it.

Comment (barentine) [reply docs/Installation/Installation.rst:25]: I also thought the last time I didn't need to add it.

Comment (David-Baddeley) [reply docs/Installation/Installation.rst:25]: hmm, I though it was there for full anaconda, but not miniconda

Comment (David-Baddeley) [reply docs/Installation/Installation.rst:25]: have added for now - we can always alter after some testing.

**Pull request merged**